tests/platform_tests/api/test_sfp.py

```diff
@@ -538,6 +547,14 @@ def test_tx_disable_channel(self, duthosts, enum_rand_one_per_hwsku_hostname, lo
                 tx_disable_chan_mask = sfp.get_tx_disable_channel(platform_api_conn, i)
                 if self.expect(tx_disable_chan_mask is not None, "Unable to retrieve transceiver {} TX disabled channel data".format(i)):
                     self.expect(tx_disable_chan_mask == expected_mask, "Transceiver {} TX disabled channel data is incorrect".format(i))
+
+            # Restore by enabling all lanes and disabling what was disabled originally
+            sfp.tx_disable_channel(platform_api_conn, i, all_lane_mask, False)
+            sfp.tx_disable_channel(platform_api_conn, i, orig_tx_disable_chan_mask, True)
+            restored_tx_disable_chan_mask = sfp.get_tx_disable_channel(platform_api_conn, i)
+            self.expect(orig_tx_disable_chan_mask == restored_tx_disable_chan_mask,
+                        "Transceiver {} restored TX disabled channel data {} is not the same as before changing tx disable".format(restored_tx_disable_chan_mask, orig_tx_disable_chan_mask))
+
```

Comment:
@harish-kalyanaraman In SONiC we dont have CLI or configuration which can disable the particular channel, so I am not sure how is this fix helping?

Reply:
The fix was to restore the sfp in the same state that we found originally before the test - as a generic rule of thumb for any good test.

Reply:
@sanmalho-git its already taken care, see the comment:- [https://github.com/sonic-net/sonic-mgmt/blob/master/tests/platform_tests/api/test_sfp.py#L529](https://github.com/sonic-net/sonic-mgmt/blob/master/tests/platform_tests/api/test_sfp.py#L529). Just needs to be taken care for QSFP-DD with 8 channels

Reply:
Have pushed a commit to remove the restoration.